cards/portfolio/addon/components/isolated.js

```diff
@@ -1,10 +1,80 @@
 import LiveIsolatedCard from 'portfolio-common/components/live-isolated-card';
 import layout from '../templates/isolated';
+import { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+import { task, waitForProperty } from 'ember-concurrency';
 
 export default LiveIsolatedCard.extend({
   layout,
+  store: service(),
+  web3: service(),
+  erc20: service(),
   isDismissed: false,
   displayGroup: 'all',
+  metamaskWallet: null,
+
+  isWeb3Loaded: computed.not('web3.isLoading'),
+
+  async init() {
+    this._super(arguments);
+
+    if (this.web3.provider && this.web3.provider.isMetaMask) {
+      await this.getMetamaskWallet.perform();
+    }
+  },
+
+  getMetamaskWallet: task(function * () {
+    yield waitForProperty(this, 'isWeb3Loaded');
+
+    let address = this.web3.address;
+    let metamaskWallet = yield this.store.findRecord('wallet', 'metamask-wallet');
+
+    this.set('metamaskWallet', metamaskWallet);
+
+    yield this.getAssetsForWallet.perform(address);
+
+    this.content.get('wallets').pushObject(this.get('metamaskWallet'));
+
+    yield this.content.save();
+  }),
+
+  getAssetsForWallet: task(function * (address) {
+    yield this.addAssetToWallet.perform(address, 'ether');
+
+    for (let token of this.erc20.tokens()) {
+      let tokenName = token.symbol.toLowerCase();
+      let balanceOf;
+
+      try {
+        balanceOf = yield this.store.findRecord(`${tokenName}-token-balance-of`, address.toLowerCase());
```

Comment:
when we have more than 100 tokens, this will be very expensive to execute. I suggest using a server side computed to determine all the tokens for an asset so you just make one call to the server to get all the tokens. This means that the erc20 service is really only consumed in the hub (you can use codegen to emit the data into the client if still need it in the browser).